contracts/core/extensions/CoreInternal.sol

```diff
@@ -50,44 +57,25 @@ contract CoreInternal is
     /* ============ External Functions ============ */
 
     /**
-     * Add a factory to the mapping of tracked factories. Can only be set by
-     * owner of Core.
-     *
-     * @param  _factory   The address of the SetTokenFactory to enable
-     */
-    function enableFactory(
-        address _factory
-    )
-        external
-        onlyOwner
-    {
-        // Mark as true in validFactories mapping
-        state.validFactories[_factory] = true;
-
-        // Add to factories array
-        state.factories.push(_factory);
-    }
-
-    /**
-     * Disable a factory in the mapping of tracked factories. Can only be disabled
-     * by owner of Core.
+     * Add or remove a factory to the mapping of tracked factories. Can only be set by
+     * owner of Core
      *
-     * @param  _factory   The address of the SetTokenFactory to disable
+     * @param  _factory   Address of the contract conforming to ISetFactory
+     * @param  _enabled   Enable or disable the factory
      */
-    function disableFactory(
-        address _factory
+    function registerFactory(
+        address _factory,
+        bool _enabled
     )
         external
         onlyOwner
     {
-        // Verify Factory is linked to Core
-        require(state.validFactories[_factory], "UNKNOWN_FACTORY");
-
-        // Mark as false in validFactories mapping
-        state.validFactories[_factory] = false;
+        state.validFactories[_factory] = _enabled;
 
-        // Find and remove factory from factories array
-        state.factories = state.factories.remove(_factory);
+        emit FactoryRegistrationChanged(
```

Comment:
This means an event will be fired on each method call, even if the state did not change. Do we want to fire an event only on explicit state change?

Reply:
Ambivalent here, the logs and the map will reflect the last state. I argue for these in particular that it's merely to track that the event was called so we can always build that array that we removed